Tests/Realm.Tests/Database/RealmDictionaryTests.cs

```diff
@@ -1073,6 +1073,56 @@ public void ObjectFromAnotherRealm_ThrowsRealmException()
             Assert.That(() => realm2.Write(() => obj2.EmbeddedObjectDictionary["foo"] = embeddedItem), Throws.TypeOf<RealmException>().And.Message.Contains("embedded object that is already managed"));
         }
 
+        [Test]
+        public void AsRealmQueryable_RaisesNotifications()
+        {
+            var obj = _realm.Write(() =>
+            {
+                return _realm.Add(new DictionariesObject());
+            });
+
+            var dogToAge = obj.ObjectDictionary;
+
+            var oldDogs = dogToAge.AsRealmQueryable().Where(d => d.Int >= 5);
+
+            var changeSets = new List<ChangeSet>();
+            var token = oldDogs.SubscribeForNotifications((sender, changes, error) =>
+            {
+                if (changes != null)
+                {
+                    changeSets.Add(changes);
+                }
+            });
+
+            for (var i = 0; i < 10; i++)
+            {
+                _realm.Write(() => dogToAge["f" + i] = new IntPropertyObject { Int = i });
+                _realm.Refresh();
+
+                if (i >= 5)
+                {
+                    Assert.That(changeSets.Count, Is.EqualTo(i - 4));
+
+                    var changeSet = changeSets.Last();
+                    Assert.That(changeSet.InsertedIndices.Length, Is.EqualTo(1));
+                    Assert.That(changeSet.DeletedIndices, Is.Empty);
+                    Assert.That(changeSet.ModifiedIndices, Is.Empty);
+
+                    Assert.That(oldDogs.ElementAt(changeSet.InsertedIndices[0]).Int, Is.EqualTo(i));
```

Comment:
Probably we should add some tests also for all the `changeSet` property (deleted, modified, newModified,...). It doesn't need to be in the same loop though

Reply:
could you elaborate what you mean by tests for `changeSet`?

Reply:
got it, this is basically line-for-line copy of the tests for sets and lists, so I imagine it'd be useful for this to be there too then.

Reply:
I suppose the reason why it's not heavily tested is the fact that we have tests for "RealmResults" in other places too, and we already expect the notifications to work properly. So this is just to be sure that the new method works property. On a second though, we could probably do the same here, and just keep it as it was